x/wasm/keeper/query_plugins.go

```diff
@@ -201,10 +202,23 @@ func BankQuerier(bankKeeper types.BankViewKeeper) func(ctx sdk.Context, request
 			}
 			return json.Marshal(res)
 		}
+		if request.AllDenomMetadata != nil {
+			denomMetadata := bankKeeper.GetAllDenomMetaData(ctx)
+			res := AllDenomMetadataResponse{
+				Metadata: denomMetadata,
+			}
+			return json.Marshal(res)
+		}
 		return nil, wasmvmtypes.UnsupportedRequest{Kind: "unknown BankQuery variant"}
 	}
 }
 
+// Response type for denom metadata queries.
+// Defining the type in wasmd instead of wasmvm allows for maximum compatibility with the cosmos-sdk Metadata type.
```

Comment:
This was a way for me to use the same type instead of a copy/pasted type in `wasmvm`. While digging into the other response types in `wasmvm`, I saw that they are not used anywhere else within the `wasmvm` codebase - they are just defined there and then only used by external packages. So, it didn't seem to be necessary to define any of these types in `wasmvm`.

However, if this is a hard requirement then I'm happy to make the change.

Reply:
@alpe moved the types all into wasmvm